src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/repositories/ecommerce/DeadLetterRepository.kt

```diff
@@ -12,17 +12,30 @@ import reactor.core.publisher.Mono
 @Repository
 interface DeadLetterRepository : ReactiveCrudRepository<DeadLetterEvent, String> {
 
-    @Query("{'insertionDate': {'\$gte': '?0','\$lte': '?1'}}", count = true)
-    fun countAllDeadLetterEventInTimeRange(startTime: String, endTime: String): Mono<Long>
+    @Query(
+        "{'insertionDate': {'\$gte': '?0','\$lte': '?1'}, 'transactionInfo.eCommerceStatus': {'\$nin': ?2}, 'transactionInfo.details.operationResult': {'\$nin': ?3} }",
+        count = true
+    )
+    fun countAllDeadLetterEventInTimeRangeWithExludeStatuses(
+        startTime: String,
+        endTime: String,
+        ecommerceStatusesToExclude: List<String>,
+        npgStatusesToExclude: List<String>
```

Comment:
```suggestion
        ecommerceStatusesToExclude: Set<String>,
        npgStatusesToExclude: Set<String>
```

please use here Set instead of List here in order to avoid putting the same state twice into the same filter